python/ray/dag/compiled_dag_node.py

```diff
@@ -657,8 +681,8 @@ def _get_node_id(self):
                 if isinstance(readers[0].dag_node, MultiOutputNode):
```

Comment:
By the way, this code seems to expect that if the driver is reading from the node, then the driver is the only one reading the node. It would be nice to lift this restriction (or at least throw a good error if true). For example, I think the following DAG would not work correctly right now:

```python
with InputNode() as inp:
  x = a.foo.bind(inp)
  y = b.foo.bind(x)
  dag = MultiOutputNode([x, y])
```
(where b and the driver both read the same output)